src/Features/Core/Portable/ExternalAccess/UnitTesting/API/UnitTestingStackTraceService.cs

```diff
@@ -0,0 +1,86 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Host;
+using Microsoft.CodeAnalysis.StackTraceExplorer;
+using System.Collections.Immutable;
+using System;
+using System.Linq;
+
+namespace Microsoft.CodeAnalysis.ExternalAccess.UnitTesting.API
+{
+    internal sealed class UnitTestingStackTraceService
+    {
+        private readonly IStackTraceExplorerService _stackTraceService;
+
+        public UnitTestingStackTraceService(HostWorkspaceServices services)
+            => _stackTraceService = services.GetRequiredService<IStackTraceExplorerService>();
+
+        public async Task<Frame?> TryParseAsync(string line, CancellationToken cancellationToken)
+        {
+            var result = await StackTraceAnalyzer.AnalyzeAsync(line, cancellationToken).ConfigureAwait(false);
+
+            if (result.ParsedFrames.Length == 0)
+            {
+                return null;
+            }
+
+            if (result.ParsedFrames.Length > 1)
+            {
+                throw new InvalidOperationException();
+            }
+
+            return new Frame(result.ParsedFrames.Single(), _stackTraceService);
```

Comment:
nit: Since this is an array can we say `result.ParsedFrames[0]`?

Reply:
Depends on https://github.com/dotnet/roslyn/pull/58082#discussion_r762364490 

If we do one result, then does multiple make sense or is that an error?

Reply:
Yes if we support only a single API here that returns multiple this is a moot point.